src/test/java/com/learning/pages/swaglabs/SwagLabsLoginPageTest.java

```diff
@@ -39,49 +40,104 @@ void shouldErrorMessageBeVisibleWhenUnsuccessfulLogin(String username, String pa
     //When
     swagLabsLoginPage.open();
     swagLabsLoginPage.performLogin(username, password);
-    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     //Then
+    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     assertEquals(actualErrorMessage, expectedErrorMessage);
   }
 
-  @Test(dataProvider = "unsuccessfulLoginData")
-  void shouldErrorMessageBeVisibleWhenLackOfPassword(String username, String password) {
+  @Test
+  void shouldErrorMessageBeVisibleWhenLackOfPassword() {
     //Given
+    String username = "standard_user";
     String expectedErrorMessage = "Epic sadface: Password is required";
     //When
     swagLabsLoginPage.open();
     swagLabsLoginPage.performLogin(username, "");
-    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     //Then
+    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     assertEquals(actualErrorMessage, expectedErrorMessage);
   }
 
-  @Test(dataProvider = "unsuccessfulLoginData")
-  void shouldErrorMessageBeVisibleWhenLackOfUsername(String username, String password) {
+  @Test
+  void shouldErrorMessageBeVisibleWhenLackOfUsername() {
     //Given
+    String password = "secret_sauce";
     String expectedErrorMessage = "Epic sadface: Username is required";
     //When
     swagLabsLoginPage.open();
     swagLabsLoginPage.performLogin("", password);
-    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     //Then
+    String actualErrorMessage = swagLabsLoginPage.getErrorText();
     assertEquals(actualErrorMessage, expectedErrorMessage);
   }
 
+  @Test
+  void shouldFailToLoginWhenLackOfCredentials() {
+    //Given
+    String expectedURL = swagLabsLoginPage.getPageURL();
+    //When
+    swagLabsLoginPage.open();
+    swagLabsLoginPage.performLogin("", "");
+    //Then
+    assertEquals(browser.getCurrentPageURL(), expectedURL);
+  }
+
+  @Test
+  void shouldNotGoToInventoryPageWithoutLogin() {
+    //Given
+    String expectedURL = swagLabsLoginPage.getPageURL();
+    //When
+    swagLabsInventoryPage.open();
+    //Then
+    assertEquals(browser.getCurrentPageURL(), expectedURL);
+  }
+
+  @Test(dataProvider = "lockedUserData")
+  void shouldNotLoginWhenLockedUser(String username, String password) {
+    //Given
+    String expectedErrorMessage = "Epic sadface: Sorry, this user has been locked out.";
+    //When
+    swagLabsLoginPage.open();
+    swagLabsLoginPage.performLogin(username, password);
+    //Then
+    assertEquals(swagLabsLoginPage.getErrorText(), expectedErrorMessage);
+  }
+
+  @Test(dataProvider = "problemUser")
+  void shouldFailToLoadPageWithProblemUserLogin(String username, String password) {
+    //When
+    swagLabsLoginPage.open();
+    swagLabsLoginPage.performLogin(username, password);
+    //Then
+    assertThrows(() -> browser.waitForPageToLoad(swagLabsInventoryPage));
+  }
 
   @DataProvider(name = "successfulLoginData")
```

Comment:
Using data providers is a great idea, I have only one suggestion, you can combine several tests by adding third parameter: `expected error message`, this way you'll be able to run most cases in a single test.